jsonschema/validators.py

```diff
@@ -356,7 +356,7 @@ def resolve_fragment(self, document, fragment):
 
         """
 
-        fragment = fragment.lstrip(u"/")
+        fragment = fragment.lstrip(u"#").lstrip(u"/")
```

Comment:
This seems likely to be misplaced overall to me (which maybe is another reason why an integration test would be good).

This method is about resolving fragments -- fragments don't start with `#`, `#` delimits a fragment, so whoever called this should have removed the `#`